src/net_processing.cpp

```diff
@@ -2410,7 +2403,7 @@ void ProcessMessage(
         UpdatePreferredDownload(pfrom, State(pfrom.GetId()));
         }
 
-        if (!pfrom.fInbound && pfrom.IsAddrRelayPeer())
+        if (!pfrom.IsInboundConn() && pfrom.IsAddrRelayPeer())
```

Comment:
agreed this can be made more clear. I took a stab at a restructure that essentially replaces `IsAddrRelayPeer` with `RelayAddrsWithConn` that checks the connection type instead of presence of `m_addr_known.` https://github.com/amitiuttarwar/bitcoin/commit/6662f5dbe1e513ea2cec9adbcc730555832f85d0. I'm still deciding how I feel about it and am interested in any feedback.

pros: 
- checking for addr relay via connection type feels more conceptually straightforward based on current conditions 
- easy to update the logic in the future 
- consistent interface being adopted with this PR 

cons: 
- the main downside is that if the initialization conditions for `m_addr_known` are ever updated, `RelayAddrWithConn` must also be updated. and would be problematic if missed by the proposer / reviewers.

Reply:
incorporated in #19724 (https://github.com/bitcoin/bitcoin/pull/19724/commits/9345f9c9d47a88987606d1421823914a4c48dbf9). I've noted this outstanding concern and am seeking opinions: https://github.com/bitcoin/bitcoin/pull/19724#issuecomment-674298044.